### DISCUSSION TITLE
Using Laravel's Paginator::defaultView breaks when using Laravel Livewire

### DISCUSSION THREAD

**Discussion by tdondich** [livewire/livewire#5179] [7. Bugs (archived)]
We have a site that utilizes Laravel::defaultView to provide a special default view for our pagination.  However, when using Livewire component with WithPagination component, it will set the Paginator defaultView to a different value during the lifecycle and then not set it back.  This is especially true when using Laravel Octane as the Paginator service is a singleton which persists across requests and if one page sets the Paginator::defaultView, then other pages fail to render or throw an exception due to Livewire's use of $this in it's paginator views.

I understand this was to try and make livewire as "transparent" as possible.  Maybe it's something where the previous value of the defaultView (and defaultSimpleView) can be saved before the component render then reset after the component render?  This would provide the best compatibility.

The use of the Paginator::defaultView and defaultSimpleView in LiveWire is present in the WithPagination trait in the initializeWithPagination() method.

**Comment by joshhanley** [comment]
@tdondich this is a duplicate of an existing, please see #5130 for details 🙂
